### ISSUE TITLE
 Mailbox for email not found.

### ISSUE THREAD

**Issue by vasikgreif** [uvdesk/community-skeleton#369]
I downloaded the latest version of UV Desk, and I'm having two issues:
- I'm unable to save the email in Email Settings - when I reload the screen, I get back the `support@localhost` default value
- when I try to create tickets from my mailbox with ` php bin/console uvdesk:refresh-mailbox support@wpify.io`, I get `Mailbox for email support@wpify.io not found.` . I do have the mailbox in the Mailboxes section, and I tried both the ID (mailbox_4251), and different emails, no luck

The yaml files look like this:

uvdesk_mailbox.yaml
```yaml
uvdesk_mailbox:
    emails: ~
        # Often Reply emails like from gmail contains extra and redundant previous mail data.
        # This data can be removed by adding delimiter i.e. specific line before each reply. 
        # delimiter: '<-- Please add content above this line -->'
        # enable_delimiter: true
    
    # Configure your mailboxes here
    mailboxes:
        mailbox_4251:
            name: WPify Support
            enabled: true

            # [SMTP] Outgoing mail server
            # Swiftmailer smtp mailer to use for sending emails through on behalf of this mailbox
            smtp_server: 
                mailer_id: mailer_9531

            # [IMAP] Incoming mail server
            # IMAP configurations to use for fetching emails from mailbox
            imap_server:
                host: 'myserver.com'
                username: support@wpify.io
                password: mypassword


```

uvdesk.yaml

```yaml

parameters:
    app_locales: en|fr|it
    
    # Default Assets
    assets_default_agent_profile_image_path: 'bundles/uvdeskcoreframework/images/uv-avatar-batman.png'
    assets_default_customer_profile_image_path: 'bundles/uvdeskcoreframework/images/uv-avatar-ironman.png'
    assets_default_helpdesk_profile_image_path: 'bundles/uvdeskcoreframework/images/uv-avatar-uvdesk.png'

    uvdesk_site_path.member_prefix: member
    uvdesk_site_path.knowledgebase_customer_prefix: customer
    
    # File uploads constraints
    # @TODO: Set these parameters via compilers
    max_post_size: 8388608
    max_file_uploads: 20
    upload_max_filesize: 2097152

uvdesk:
    site_url: 'support.wpify.io'
    upload_manager:
        id: Webkul\UVDesk\CoreFrameworkBundle\FileSystem\UploadManagers\Localhost
    
    support_email:
        id: support@wpify.io
        name: WPify.io Support
        mailer_id: mailer_9531
    
    # Default resources
    default:
        ticket:
            type: support
            status: open
            priority: low
        templates:
            email: mail.html.twig

```

swiftmailer.yaml

```yaml
swiftmailer:
    default_mailer: mailer_9531
    mailers:
        mailer_9531:
            transport: smtp
            username: support@wpify.io
            password: password
            host: myserver.com
            port: 465
            encryption: ssl
            auth_mode: login
            sender_address: support@wpify.io
            delivery_addresses: ['support@wpify.io']
            disable_delivery: false


```
Any idea what to do?

Thanks!

**Comment by papnoisanjeev** [comment]
@vasikgreif 

May be due to your files under **config/packages** folder have no **read/write** permission so due to this you are facing issue.

Please enable **read/write** permission by **chmod -R 775 .path_upto_packages_folder**

**Comment by PeopleInside** [comment]
@vasikgreif have you resolved?

**Comment by vasikgreif** [comment]
No, i gave up, too many issues trying to make this work.

**Comment by PeopleInside** [comment]
Umh... the two issue you reported is well know for me, I had the same issue so is a common problem that facing new users that decide to setup UVdesk.

I'm not a developer so I cannot help much. I'm a UVdesk user that has choose to use UVdesk.
I'm sorry about reading you still have issue.

UVdesk is a great Open Source helpdesk that give many great feature and a very nice responsiveness so I suggest to still try if you did not found a better solution.

Maybe if you update developers @papnoisanjeev  you will see that you can find a nice team ready to help you.
I did not remember how I solved this issue...

Are you sure to not use special characters in the email password?
What version of PHP are you running? I don't think this issue is related to PHP as I had also when I tried to setup UVdesk but finally I was able to resolve...

Le's see if @papnoisanjeev has other suggestions. You are not the only one that has this issue on my opinion.

**Comment by vasikgreif** [comment]
Thanks for the update. I'm experienced PHP developer myself, but the lack of a good documentation makes it really difficult to setup UV Desk. I'm fine with editing config files, but I'd expect a clear documentation on the steps needed.

I'll probably come back to UV Desk once the project matures a bit more..:)

**Comment by PeopleInside** [comment]
Without promise nothing I will try to test and help with the two issues you are facing.
Testing this ( if I find time and find resource on my server to do ) can help on improve guide for users.

**Comment by PeopleInside** [comment]
OK @vasikgreif  for resolve the first issue

> I'm unable to save the email in Email Settings - when I reload the screen, I get back the support@localhost default value

After setup 

    Swift Mailer
    MailBoxes

I tried to change the support@localhost but also if the page was saved at the refresh the settings is not saved.

You need run the following SSH command on the root of uvdesk

**php bin/console c:c**

After come back on your ticket system operator and refresh the page with email settings ad you will see now your correct email address.

A note for @papnoisanjeev 
I'm installing a testing install for help the user here. My install is made by Softacolous scrip installer with PHP 7.4

Doing the mentioned SSH command I get the following error warning:

```
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 351
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 385
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 394
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 408
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 416
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 451
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 529
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 559
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 637
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 638
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 640
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 678
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 679
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 681
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 835
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 835
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 839
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 839
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 843
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 843
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 847
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /uvdesk/vendor/uvdesk/core-framework/Utils/HTMLFilter.php on line 847
```

I will now test the email mailbox ticket creation and let you know.

**Comment by PeopleInside** [comment]
Sento a test email to the support email setup and run the SSH command
`php bin/console uvdesk:refresh-mailbox emailaddress@domain.ext`

Ticket opened.
Also your second issue reported should be resolved.

**Comment by PeopleInside** [comment]
SSH give:

**Comment by PeopleInside** [comment]
SSH Give:

```
 1. Processing uvdesk mailbox configuration.

 2. Opening imap stream...

 3. IMAP stream opened.

 4. Fetching Email collection since 22 March 2021.

 5. Total fetched email -> 1

 6. Starting to convert Emails into Tickets ->
```

But ticket is not created and email is set as read.
This is another common issue on UVdesk. I'm trying to resolve but it's really hard.

First of all need set the correct domain in /uvdesk/config/packages/uvdesk.yaml removing port 8000 like

site_url: hepdesk.website.ext

Here I still be unavle to see ticket created. I will update as soon I can understand why this still happening.

**Comment by PeopleInside** [comment]
OK for resolve the fact ticket are not opened I take a look at ticket I created in UVdesk on 07-11-20
I was near to leave all and tell you I'm unable to resolve this issue that I have resolved on my website many months ago.

The issue was the site URL should have public so my site URL was

site_url: hepdesk.website.ext

and did not work, changed in

site_url: hepdesk.website.ext/uvdesk/public

than on SSH you need do in uvdesk root:

**php bin/console c:c**

thank now you can send an email to your support email address and SSH

php bin/console uvdesk:refresh-mailbox emailaddress@domain.ext

I will try to create a guide in the UVdesk forum where also UVdesk team can improve my post or do a different official guide.

Your issue can be closed once you can check all works fine.
I tested for you and resolved the two issue reported by you.

It's all because there is not a clear guide, your are right!

**Comment by abielikesu** [comment]
@vasikgreif I had the same issue and manage to resolve it.

TLDR; In the **config/packages/uvdesk_mailbox.yaml** file, the host syntax has to be in the format below:
```
            imap_server:
                host: '{imap.myserver.com:993/imap/ssl}INBOX'
                username: myuser@mycomain.com
                password: mypassword

```
The code that opens the imap inbox is using the [imap_open](https://www.php.net/manual/en/function.imap-open.php) function. 
Look at the different examples and choose the one that suits your setup. In my case it was:

**Comment by vasikgreif** [comment]
Well, unless the setup will work out of the box, without having to edit config files and go to Github to make basic functionality working, I still consider UVDesk a great project, which is unfortunately not mature enough to be actually used in production.

**Comment by PeopleInside** [comment]
@papnoisanjeev  @vasikgreif  tis issue can maybe be closed?
I gived the solution and asked to think maybe to improve in another different topic https://github.com/uvdesk/community-skeleton/issues/382

I think no active issue are present on this topics except the user request to implement documentation and to be able to use just the UI, something I wrote in the new issue.

**Comment by Ntworkadmin** [comment]
we install uvdesk on ubuntu with docker and followed https://shownotes.opensourceisawesome.com/uvdesk/ this everything working fine except incoming emails. so any incoming email example@company.com not creating ticket automatically and when we create new ticket client get email notification and whenever agent reply they get email but if they reply is coming example@company.com but not showing up on uvdesk system as reply any help would be appreciated

**Comment by PeopleInside** [comment]
@Ntworkadmin look at the forum: https://forums.uvdesk.com/topic/1988/how-to-debug-errors-and-email-fetching-errors

**Comment by conniviatech16** [comment]
> SSH Give:
> 
> ```
>  1. Processing uvdesk mailbox configuration.
> 
>  2. Opening imap stream...
> 
>  3. IMAP stream opened.
> 
>  4. Fetching Email collection since 22 March 2021.
> 
>  5. Total fetched email -> 1
> 
>  6. Starting to convert Emails into Tickets ->
>  =============================================
> =============================================
> 
> Converting email 1 out of 1.
> 
> Mailbox refreshed successfully !!!
> ```
> 
> But ticket is not created and email is set as read. This is another common issue on UVdesk. I'm trying to resolve but it's really hard.
> 
> First of all need set the correct domain in /uvdesk/config/packages/uvdesk.yaml removing port 8000 like
> 
> site_url: hepdesk.website.ext
> 
> Here I still be unavle to see ticket created. I will update as soon I can understand why this still happening.

its working my ..i have cpanel can you help how can I set cron job in cpanel .. i have set, but not getting email .. when I run via terminal all mail got fatched

**Comment by PeopleInside** [comment]
@conniviatech16 you can ask help on the UVdesk forum if you need.
Writing here you generate an email to all participant and this is annoying for anyone.

If you don't want to register to the UVdesk community you can try to ask [here](https://github.com/uvdesk/community-skeleton/discussions)
